woocommerce/woocommerce-gutenberg-products-block
# CollectionFilters: Hydrate Collection Data for inner filter blocks

## What

In this PR, we hydrate the collection data API request for all inner filter blocks from the `CollectionFilters` block, then passing down the response through context.

Fixes #10984

## Why

By collecting the information of inner filter blocks and hydrating the API request from the parent block, we only need to hydrate one time for all inner filter blocks.



## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Update the `$colleciton_data_params_mapping` property as follow
```
	private $collection_data_params_mapping = array(
		'calculate_price_range'         => 'woocommerce/price-filter',
		'calculate_stock_status_counts' => 'woocommerce/stock-filter',
		'calculate_attribute_counts'    => 'woocommerce/attribute-filter',
		'calculate_rating_counts'       => 'woocommerce/rating-filter',
	);
```
2. Update the `render()` method of `src/BlockTypes/PriceFilter.php` as follow:
```
	protected function render($attributes, $content, $block) {
		error_log( 'Block name: ' . $block->name );
		error_log( print_r( $block->context, true ) );
		return $content;
	}
```
4. On a new page, add the `Product Collection` block.
5. Then add `Collection Filters` block to the block above.
6. Add Price filter to the `Collection Filters` block.
7. Visit the page on the front end.
8. Check the log, see the data includes collection data response for price and stock, something similar to this

```
[14-Sep-2023 11:43:26 UTC] Block name: woocommerce/price-filter
[14-Sep-2023 11:43:26 UTC] Array
(
    [isCollectionFiltersInnerBlock] => 1
    [collectionData] => Array
        (
            [price_range] => stdClass Object
                (
                    [min_price] => 0
                    [max_price] => 25000
                    [currency_code] => USD
                    [currency_symbol] => $
                    [currency_minor_unit] => 3
                    [currency_decimal_separator] => ,
                    [currency_thousand_separator] => .
                    [currency_prefix] => $
                    [currency_suffix] => 
                )

            [attribute_counts] => 
            [rating_counts] => 
            [stock_status_counts] => 
        )

)
```
9. Wrap the Price Filter block in a Group block. Visit the page on the frontend, see the new log entries indicates the logic still work with nested block.


## Screenshots or screencast



| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Add suggested changelog entry here.
